### PR TITLE
refactor selection of linter from advice code

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Our linters generate advices with advice codes that may vary depending on their analysis.
Since fixers are typically implemented as linters (see `FromTableSqlLinter`, `SparkSqlPyLinter`, `SparkTableNamePyLinter`), we need them to be used for any advice code generated during linting (that they can actually fix)
This is not achievable with the current implementation, because the `name` property can only return a single code (plus its purpose is obfuscated).
This PR fixes the above by renaming `name` and having it return multiple values

### Linked issues
None

### Functionality
None

### Tests
- [x] ran unit tests

### CODE REVIEW

File: src/databricks/labs/ucx/source_code/linters/context.py [hunk 1]
```diff
@@ -83,14 +83,14 @@ def fixer(self, language: Language, diagnostic_code: str) -> Fixer | None:
         if language not in self._fixers:
             return None
         for fixer in self._fixers[language]:
-            if fixer.name == diagnostic_code:
+            if diagnostic_code in fixer.supported_advice_codes:
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/linters/context.py:86]: Diagnostic or advice?

**Review by JCZuurmond — COMMENTED** [review]
One comment to clarify

File: src/databricks/labs/ucx/source_code/linters/context.py (hunk 1 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/linters/context.py:86]: it's the code found in Advice.code so the latter, maybe we need to also rename the parameter

**Review by JCZuurmond — APPROVED** [review]
lgtm

**Comment by ericvergnaud** [comment]
@nfx not strategic, closing as discussed

**Pull request closed without merging**